src/raftstore/store/engine.rs

```diff
@@ -223,24 +263,28 @@ impl Peekable for DB {
 }
 
 impl Iterable for DB {
-    fn new_iterator(&self, upper_bound: Option<&[u8]>, fill_cache: bool) -> DBIterator {
+    fn new_iterator(&self, iter_opt: IterOption) -> DBIterator {
         let mut readopts = ReadOptions::new();
-        readopts.fill_cache(fill_cache);
-        if let Some(key) = upper_bound {
-            readopts.set_iterate_upper_bound(key);
+        readopts.fill_cache(iter_opt.fill_cache);
+        readopts.set_total_order_seek(match iter_opt.seek_mode {
```

Comment:
I see 4 places use the same way to set the order seek, this is redundant. Seem that you can extract a function like `total_order_seek_used(&self) -> bool` for the iterator option.